ConnectionFactory.php

```diff
@@ -40,22 +43,23 @@ public function __construct(array $typesConfig)
      */
     public function createConnection(array $params, Configuration $config = null, EventManager $eventManager = null, array $mappingTypes = [])
     {
+        $connection = DriverManager::getConnection($params, $config, $eventManager);
```

Comment:
instantiating the connection before types are registered in the registry creates a risk that mapping types defined through the `\Doctrine\DBAL\Types\Type::getMappedDatabaseTypes` might not be registered properly in the platform (that happens when the platform is instantiated, but your patch does it after registering the first type).